superset/models/core.py

```diff
@@ -691,27 +731,30 @@ def _log_query(sql: str) -> None:
         with self.get_raw_connection(catalog=catalog, schema=schema) as conn:
```

Comment:
WDYT about introducing the `temporarily_disconnect_db` logic inside `get_raw_connection`, if possible, could introduce several benefits, for example would make sure that all analytics db queries would obey the disconnect, easier to refactor in the future

Reply:
That was my original plan, but some code like Database.get_df() doesnt use get_raw_connection. Also looked at get_sqla_engine and other areas in DbEngineSpec. There's a fair amount of indirection and a deep, conditional call stack around analytics queries... One thing I found is that with context managers like Database.get_sqla_engine, we offer a lease on a connection and there's no guarantees that the code using it won't need a metadata connection...

Reply:
Would be there refactor analytics DB access to got through a single focal point. I'd say a the logic related to external data access in core.models.Database and in sql_lab.py and elsewhere should be brought in DbEngineSpec. Make a lot of of the logic private to that package and have the rest of the codebase use higher level abstractions like get_df or get_data.

Reply:
> WDYT about introducing the temporarily_disconnect_db logic inside get_raw_connection

Unfortunately I don't think that it is possible, given that `get_raw_connection` is context manager, and that it's possible for the caller to use the metadata database in the context provided by the context manager...